File: pec-observation-app/ios-app/Sources/PECObservation/Domain/Entities/KnowledgeDocument.swift
import Foundation

public enum DocumentType: String, Codable, CaseIterable {
    case seducPolicy      = "seduc_policy"
    case curriculumGuide  = "curriculum_guide"
    case evaluationRubric = "evaluation_rubric"
    case feedbackExample  = "feedback_example"
    case other            = "other"

    public var displayName: String {
        switch self {
        case .seducPolicy:      return "Política SEDUC"
        case .curriculumGuide:  return "Guia Curricular"
        case .evaluationRubric: return "Rubrica de Avaliação"
        case .feedbackExample:  return "Exemplo de Feedback"
        case .other:            return "Outro"
        }
    }
}

public struct KnowledgeDocument: Identifiable, Codable, Equatable {
    public let id: String
    public let title: String
    public let documentType: String
    public let sourceFilename: String
    public let description: String?
    public let chunkCount: Int
    public let uploadedBy: String
    public let isActive: Bool
    public let createdAt: String

    public var documentTypeDisplay: String {
        DocumentType(rawValue: documentType)?.displayName ?? documentType
    }

    enum CodingKeys: String, CodingKey {
        case id, title, description, isActive = "is_active", createdAt = "created_at"
        case documentType = "document_type", sourceFilename = "source_filename"
        case chunkCount = "chunk_count", uploadedBy = "uploaded_by"
    }
}

File: pec-observation-app/android-app/app/src/main/kotlin/br/gov/educacao/sp/pecobservation/domain/entities/KnowledgeDocument.kt
package br.gov.educacao.sp.pecobservation.domain.entities

import java.time.Instant

enum class DocumentType(val displayName: String) {
    SEDUC_POLICY("Política SEDUC"),
    CURRICULUM_GUIDE("Guia Curricular"),
    EVALUATION_RUBRIC("Rubrica de Avaliação"),
    FEEDBACK_EXAMPLE("Exemplo de Feedback"),
    OTHER("Outro");

    companion object {
        fun fromRaw(raw: String) = entries.firstOrNull { it.name.lowercase() == raw } ?: OTHER
    }
}

data class KnowledgeDocument(
    val id: String,
    val title: String,
    val documentType: String,
    val sourceFilename: String,
    val description: String?,
    val chunkCount: Int,
    val uploadedBy: String,
    val isActive: Boolean,
    val createdAt: String,
) {
    val documentTypeDisplay: String
        get() = DocumentType.fromRaw(documentType).displayName
}

enum class StyleTone(val displayName: String) {
    FORMAL("Formal"),
    CONSTRUCTIVE("Construtivo"),
    DIRECT("Direto"),
    SUPPORTIVE("Acolhedor");
}

data class FeedbackStyle(
    val id: String,
    val name: String,
    val description: String,
    val tone: String,
    val templatePrompt: String,
    val exampleStrengths: List<String>,
    val exampleImprovements: List<String>,
    val isActive: Boolean,
    val isDefault: Boolean,
    val createdAt: String,
) {
    val toneDisplay: String
        get() = StyleTone.entries.firstOrNull { it.name.lowercase() == tone }?.displayName ?: tone
}
